File: src/codes/Users_Manual/group.cpp
//~#include<autd3.hpp>
//~int main() {
//~using namespace autd3;
//~const auto x = 0.0;
//~const auto y = 0.0;
//~const auto z = 0.0;
autd3::Group(
    [](const Device& dev) -> std::optional<const char*> {
      if (dev.idx() == 0) {
        return "null";
      } else if (dev.idx() == 1) {
        return "focus";
      } else {
        return std::nullopt;
      }
    },
    std::unordered_map<const char*, std::shared_ptr<driver::Datagram>>{
        {"focus",
         std::make_shared<Focus>(Focus(Point3(x, y, z), FocusOption{}))},
        {"null", std::make_shared<Null>()}});
//~return 0; }

File: src/codes/Users_Manual/group.rs
# use autd3::prelude::*;

# use std::collections::HashMap;
# fn main() {
#     let x = 0.;
#     let y = 0.;
#     let z = 0.;
#     let _ = 
Group::new(
    |dev| match dev.idx() {
        0 => Some("focus"),
        1 => Some("null"),
        _ => None,
    },
    HashMap::from([
        (
            "focus",
            BoxedDatagram::new(Focus {
                pos: Point3::new(x, y, z),
                option: Default::default(),
            }),
        ),
        ("null", BoxedDatagram::new(Null {})),
    ]),
);
# }